src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -769,7 +769,7 @@ trait TypeDiagnostics extends splain.SplainDiagnostics {
         )
         def warningIsOnFor(s: Symbol) =
           if (!s.isImplicit) settings.warnUnusedExplicits
-          else if (!s.isSynthetic) settings.warnUnusedImplicits
+          else if (!s.isSynthetic) settings.warnUnusedImplicits && s.info.members.reverseIterator.exists(m => !m.isPrivate && !m.isConstructor && !isUniversalMember(m))
           else settings.warnUnusedSynthetics
```

Comment:
this means it does not apply to context bound, which is why the test result t12591 changed on the other PR.

It should not matter whether I express it as bound, `[A: Marker]` or `(implicit marker: Marker[A])`.